include/idf/EthernetWingMan.hh

```diff
@@ -0,0 +1,42 @@
+/*
+PURPOSE:
+LIBRARY DEPENDENCIES: (
+(idf/EthernetWingMan.cpp)
+)
+*/
+
+/**
+ * @trick_parse{everything}
+ * @trick_link_dependency{idf/EthernetWingMan.cpp}
+ */
+
+#ifndef ETHERNET_WING_MAN_HH
+#define ETHERNET_WING_MAN_HH
+
+#include "idf/EthernetDevice.hh"
+#include "idf/WingMan.hh"
+
+namespace idf {
+
+/** ETHERNET WingMan joystick */
+class EthernetWingMan : public EthernetDevice, public virtual WingMan {
+
+    public:
+
+    /** constructor */
+    EthernetWingMan();
+
+    virtual void decode(const std::vector<unsigned char>& data);
+
+    virtual void setUDP() { EthernetDevice::setUDP("Hello"); }
```

Comment:
oversight thanks.

Reply:
Actually, remembering now: The idea is that the implementing class should define the connection greeting, but it can't necessarily be hardcoded or determined on instatiation.

For Wingman and Extreme3dPro, they were just put together to assist with testing, so it doesn't need anything specific, just somethings. But for our incoming device it needs a message with a range of axes and a binary CRC. I want the device implementation to be able to customize the greeting, but also to provide a setUDP() with no arguments so that the idf_config.py can just call that.